File: palindromeNumbers.js
/*
Palindrome Numbers:
? A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
! Find the largest palindrome made from the product of two 3-digit numbers.
*/

const palindromeNumbers = (start, stop) => {
    let palindromesArr = []
    for (let i = start; i < stop; i++) {
        for (let n = start; n < stop; n++) {
            let product = i * n
            let str = product.toString()
            let arr = str.split("")
            let reversedArr = arr.reverse()
            let reversedStr = reversedArr.join("")
            if (str === reversedStr) {
                palindromesArr.push(product)
            }
        }
    }
    const largestPalindrome = Math.max(...palindromesArr)
    return largestPalindrome
}

const answer = palindromeNumbers(100, 1000)
console.log(answer)

File: palindrome_numbers.py
# pylint: disable=missing-module-docstring, disable=line-too-long, consider-using-enumerate, missing-function-docstring, invalid-name
"""
Palindrome Numbers:
? A palindromic number reads the same both ways. The largest palindromemade from the product of two 2-digit numbers is 9009 = 91 x 99.

! Find the largest palindrome made from the product of two 3-digit numbers.
"""


def palindrome_numbers(start, stop):
    palindromes_list = []
    for i in range(start, stop):
        for n in range(start, stop):
            product = i * n
            if str(product) == str(product)[::-1]:
                palindromes_list.append(product)
            else:
                continue

    palindromes_list.sort()
    answer = max(palindromes_list)
    return answer


found_palindrome = palindrome_numbers(100, 1000)

print(found_palindrome)
